sycl/include/sycl/ext/oneapi/experimental/fixed_size_group.hpp

```diff
@@ -112,10 +112,23 @@ template <size_t PartitionSize, typename ParentGroup> class fixed_size_group {
   }
 
 protected:
+
+#if defined (__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
+  sub_group_mask Mask;
+#endif
```

Comment:
I'm not sure if this is a problem, but I want to ask the question and see what @intel/llvm-reviewers-runtime think...

If this is guarded like this, `sizeof(fixed_size_group)` will be different for different backends.  Does that matter?

Reply:
Yep this is one of the main questions I have regarding the design. @sergey-semenov do you have an opinion on this?
If it is an issue then is it OK to simply remove the guards?

btw I've now got working impls for all non-uniform group algorithms and finalised the design in the cuda backend. So as soon as this PR is merged and the issues like the above are resolved, I can add the algorithms accordingly.

Reply:
@Pennycook I'm going to assume that this will stay as it is at the moment so I can finalize the algorithms implementation that I'm doing on top of this PR: I've tried to ask around and the consensus seems to be to leave it as it is.

Reply:
@JackAKirk Sounds good to me.  Sorry for holding things up.

Reply:
No worries, thanks!

Reply:
I've finished the algorithm implementation part on top of this PR here: https://github.com/intel/llvm/compare/sycl...JackAKirk:llvm:cuda-non-uniform-algs

Everything except the reduce/scans uses the same impl for all non-uniform groups. Reduce also uses the same impl for all group types on sm80 for special `IsRedux` types/ops pairs.
Otherwise reduce/scans have two impl categories: 
1.fixed_size_group
2.opportunistic_group, ballot_group, (and tangle_group once it is supported) all use the same impls.

I've added a small additional commit in this PR to optimize `IdToMaskPosition`. I think this PR is ready for review. Once it is merged I'll open a PR for https://github.com/intel/llvm/compare/sycl...JackAKirk:llvm:cuda-non-uniform-algs

Reply:
Done